bin/p2-exec/main.go

```diff
@@ -32,6 +32,7 @@ var (
 		"Specifies an extra environment KEY=VALUE pair to set for the process. May be used multiple times. Takes precedence over --env if there are conflicts",
 	).StringMap()
 	launchableName = kingpin.Flag("launchable", "The key in $PLATFORM_CONFIG_PATH containing the cgroup parameters.").Short('l').String()
+	readyFile      = kingpin.Flag("ready-file", "Check for the presence of a ready-to-run file before execing its argument").String()
```

Comment:
nit: Calling this a "ready file" seems strange. To me, without the context you have right now, the name doesn't convey the meaning of what it does. I would suggest something more straightforward that doesn't introduce a new term:
```
  --require-file FILENAME
      If given, require the presence of the named file before running the command.
```